### PR TITLE
Cat requirements.txt file to aid with debugging

### PR DESCRIPTION
**Pull request by Kami**
This will allow us to see the final generated `requirements.txt` file and hopefully aid us with debugging.

### CODE REVIEW

**Comment by lakshmi-kannan** [comment]
Naice!

**Comment by Kami** [comment]
I also pushed a change so we now fix `st2client` dependency to a specific version we build and we dynamically inject this dependency with a full version identifier to st2 package `requirements.txt`.

**Comment by armab** [comment]
Nice debugging stuff :+1: 


From the build log:
https://circleci.com/gh/StackStorm/st2-packages/650

```py
[wheelhouse: st2]      cat requirements.txt
[wheelhouse: st2]      # Don't edit this file. It's generated automatically!
[wheelhouse: st2]      apscheduler<3.1,>=3.0.5
[wheelhouse: st2]      bcrypt
[wheelhouse: st2]      bencode<1.1,>=1.0
[wheelhouse: st2]      eventlet<0.19,>=0.18.4
[wheelhouse: st2]      git+https://github.com/Kami/logshipper.git@stackstorm_patched#egg=logshipper
[wheelhouse: st2]      git+https://github.com/StackStorm/fabric.git@stanley-patched#egg=fabric
[wheelhouse: st2]      git+https://github.com/StackStorm/pecan.git@st2-patched#egg=pecan
[wheelhouse: st2]      git+https://github.com/StackStorm/python-mistralclient.git#egg=python-mistralclient
[wheelhouse: st2]      git+https://github.com/StackStorm/st2-auth-backend-flat-file.git@master#egg=st2-auth-backend-flat-file
[wheelhouse: st2]      git+https://github.com/StackStorm/st2-auth-backend-pam.git@master#egg=st2-auth-backend-pam
[wheelhouse: st2]      gitpython==0.3.2.1
[wheelhouse: st2]      gunicorn<20.0,>=19.4.5
[wheelhouse: st2]      ipaddr
[wheelhouse: st2]      jinja2
[wheelhouse: st2]      jsonpath-rw>=1.3.0
[wheelhouse: st2]      jsonschema>=2.3.0
[wheelhouse: st2]      kombu
[wheelhouse: st2]      lockfile<0.11,>=0.10.2
[wheelhouse: st2]      mongoengine<0.9,>=0.8.7
[wheelhouse: st2]      networkx==1.10
[wheelhouse: st2]      oslo.config<1.13,>=1.12.1
[wheelhouse: st2]      oslo.utils<3.1.0
[wheelhouse: st2]      paramiko<1.16,>=1.15.2
[wheelhouse: st2]      passlib<1.7,>=1.6.2
[wheelhouse: st2]      prettytable
[wheelhouse: st2]      pyinotify<=0.10,>=0.9.5
[wheelhouse: st2]      pymongo<3.0
[wheelhouse: st2]      python-dateutil
[wheelhouse: st2]      python-gnupg<0.4,>=0.3.7
[wheelhouse: st2]      python-json-logger
[wheelhouse: st2]      pyyaml<4.0,>=3.11
[wheelhouse: st2]      requests[security]<3.0,>=2.7.0
[wheelhouse: st2]      retrying
[wheelhouse: st2]      semver>=2.1.2
[wheelhouse: st2]      six==1.9.0
[wheelhouse: st2]      st2actions
[wheelhouse: st2]      st2api
[wheelhouse: st2]      st2auth
[wheelhouse: st2]      st2common
[wheelhouse: st2]      st2exporter
[wheelhouse: st2]      st2reactor
[wheelhouse: st2]      st2stream
[wheelhouse: st2]      stevedore<1.8,>=1.7.0
[wheelhouse: st2]      tooz==1.20.0
[wheelhouse: st2]      virtualenv<14.0,>=13.1.2
[wheelhouse: st2]      st2client==
```

and error: 

```py
RequirementParseError: Expected version spec in st2client== at ==
```

^^ cc @manasdk why PR status with log is useful at all and relates to developer productivity.
It literally saved hundreds of working hours for several persons.

Removing PR builds will complicate debugging and make feedback loop worse.

**Comment by manasdk** [comment]
> ^^ cc @manasdk why PR status with log is useful at all and relates to developer productivity.
It literally saved hundreds of working hours for several persons.

> Removing PR builds will complicate debugging and make feedback loop worse.

@armab This is a bad one to point out to me. Reason -

1. developer should see this problem locally even before it shows up here. This is sort of like saying should not have to run any unit tests or builds locally before merging to master. That completely incorrect and is 100% developer responsibility. If you are saying some of these builds cannot be run locally then I suspect we have a problem with a build pipeline which cannot be reproduced locally - I doubt you are saying so.
2. If developer does want to run builds on circleci then push the branch to github and manually trigger via https://circleci.com/docs/api or (circle_ci pack)[https://github.com/StackStorm/st2contrib/blob/master/packs/circle_ci/actions/run_build.yaml] a build. This can be a very conscious choice and done as per requirement i.e. the 100s of working hours of several persons can still be saved.

**Comment by armab** [comment]
@manasdk Talking in numbers & examples.
Current CI for this repo runs 5 containers to build & test packages under: 
* `debian wheezy`
* `debian jessie`
* `ubuntu trusty`
* `el6`
* `el7`

Overall builds takes `8 mins` in CI (`5 distros` in parallel).

As @lakshmi-kannan stated, building `st2-packages` locally (see [README.md](https://github.com/StackStorm/st2-packages#usage) for instructions) for his laptop it takes `half of hour` to build+autotest simple `wheezy` packages.
Considering that he would like to catch bugs before even opening PR - he tests under all 5 platforms. It will take `2+ hours` to build/test one single `st2-packages` change, while via CI it's just 8 minutes.

------

If the concern only about `st2` repo - no PR build & status for packages enabled in that repo.

For all other repos which should produce packages - PR builds/statuses are enabled.
If there is intent to disable PR builds for other repos (*because it can DDOS CI*), - that's change I can't agree with.

**Comment by manasdk** [comment]
@armab 6 min locally

**Comment by manasdk** [comment]
@armab let's stop talking about this here. It's distracting to the PR and pretty much irrelevant. Ping me offline to continue if you wish to hear.

File: packages/st2/Makefile [hunk 1]
```diff
@@ -66,6 +66,19 @@ ifeq ($(DEBIAN),1)
...
+
+# Note: We want to dynamically inject "st2client" dependency. This way we can
+# pin it to the version we build so the requirement is satisfied by locally
+# built wheel and not version from PyPi
+inject-deps: .stamp-inject-deps
+.stamp-inject-deps:
+	# Note: We dynamically obtain the version, this is required because dev
+	# build versions don't store correct version identifier in __init__.py
+	# and we need setup.py to normalize iz
+	$(eval $ST2PKG_NORMALIZED_VERSION :=$(shell python setup.py --version))
```
Comment (dennybaa) [inline packages/st2/Makefile]: move this up with the bro lines ^^, try to avoid eval.

**Pull request merged**